### ISSUE TITLE
Due to Snapshot issue, the app dies after some time.

### ISSUE THREAD

**Issue by Ktaewon** [nanihadesuka/LazyColumnScrollbar#14]
I found an issue where the initial installation might be fine, but the app just quits after a while. This issue may appear immediately after installing the app, or may appear after a random amount of time.

The sample app in the repository also shows the same problem.

```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.*****.*********, PID: 32681
    java.lang.IllegalStateException: Reading a state that was created after the snapshot was taken or in a snapshot that has not yet been applied
```

I confirmed that the part I found about this issue appeared in the part where `translationY` was calculated and applied.
Like this line: `translationY = constraints.maxHeight.toFloat() * normalizedOffsetPosition`

Digging deeper, it is expected that the problem occurs in the part where the normalizedThumbSizeReal is accessed in the `offsetCorrection` function.

**Comment by nanihadesuka** [comment]
Yes, I found this issue myself when adding the UI unit test. The error is clear but the solution not so much. Will investigate more.

**Comment by nanihadesuka** [comment]
@Ktaewon 
> Digging deeper, it is expected that the problem occurs in the part where the normalizedThumbSizeReal is accessed in the offsetCorrection function

Why is expected to fail here? by official documentation that should be valid behaviour

**Comment by nanihadesuka** [comment]
Did fix but will probably generate more recompositions from now on.

**Comment by Ktaewon** [comment]
> @Ktaewon
> 
> > Digging deeper, it is expected that the problem occurs in the part where the normalizedThumbSizeReal is accessed in the offsetCorrection function
> 
> Why is expected to fail here? by official documentation that should be valid behaviour

 First of all, thank you so much for solving the issue quickly. I found it by checking the code line by line for the part I checked in that section.

 I'm glad you solved the problem, but after analyzing it further today, I found that the part that calculates the stickyHeader in my case was the cause, and when I deleted the code, it worked normally.
 I will check again for the modified code. Thank you.

_In addition, thanks to your good code, I am modifying the code to work for LazyVerticalGrid as well. If it works normally, is it possible to add it?_
